packages/editor/src/components/block-list/block.js

```diff
@@ -93,6 +93,12 @@ export class BlockListBlock extends Component {
 		if ( this.props.isSelected && ! prevProps.isSelected ) {
 			this.focusTabbable( true );
 		}
+
+		// When triggering a multi-selection,
+		// move the focus to the wrapper of the first selected block.
+		if ( this.props.isFirstMultiSelected && ! prevProps.isFirstMultiSelected ) {
+			this.wrapperNode.focus();
```

Comment:
Yes, it's meant to occur instead of `focusTabbable` and yes `focusTabbable` won't occur here because the `isSelected` is false when the block is multiselected.

I'm not sure about the assertive technologies though, it's very tied with multi-selection, not sure how multi-selection is announced.